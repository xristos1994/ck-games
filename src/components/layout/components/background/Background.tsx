import React, { FC, ReactElement } from "react";
import { classnames } from "@utils/component-utils";

const styles = require("./styles.module.css");

const starClasses: Record<string, string> = {
  smallStar: "smallStar",
  mediumStar: "mediumStar",
  bigStar: "bigStar",
};

const Background: FC = (): ReactElement => {
  return (
    <div className={styles.backgroundContainer}>
      {Array.from(" ".repeat(screen.width / 10)).map(() => {
        const randomNumber = Math.random();
        const starClass =
          randomNumber < 0.1
            ? starClasses.bigStar
            : randomNumber < 0.3
            ? starClasses.mediumStar
            : starClasses.smallStar;

        return (
          <div
            className={classnames(styles.star, styles[starClass])}
            style={{
              left: Math.random() * screen.width,
              top: Math.random() * screen.height,
            }}
          ></div>
        );
      })}
    </div>
  );
};

export { Background };
