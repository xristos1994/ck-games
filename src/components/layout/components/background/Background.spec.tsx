import React from 'react';
import renderer from 'react-test-renderer';
import { Background } from './Background';

describe('components/Background', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Background />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
