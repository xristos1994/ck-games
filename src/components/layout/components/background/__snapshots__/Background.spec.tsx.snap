// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Background renders correctly 1`] = `
<div
  className="backgroundContainer"
>
  <div
    className="stars1"
  />
  <div
    className="stars2"
  />
  <div
    className="stars3"
  />
</div>
`;
